### ISSUE TITLE
[BUG] tsconfig paths for routes not being consumed correctly

### ISSUE THREAD

**Issue by Bengejd** [compodoc/compodoc#1108]
##### **Overview of the issue**

Compodoc isn't taking the tsconfig baseURL into account when generating routes, so the documentation for routing is completely broken when using shorthand imports (a must-have in larger projects).

It produces the following error for all route.ts files: 
`reason: FileNotFoundError: File not found: C:/Users/Benge/Documents/GitHub/@myclient/src/app/app/core/components/patients-list/patients-list.routes.ts`

And as you can see, compodoc is generating a second `app` path, where it should only have one.

This doesn't seem to be an issue for any other file type, such as components, modules, services, etc... only routes. This means it's not an issue of "it can't be done", just an issue of "It's not currently being done, for routes".

To generate any documentation in my project currently, without a massive refactoring (which I'm not even sure I could justify to my boss), is to disable route generation via `--disableRoutesGraph`

##### **Operating System, Node.js, npm, compodoc version(s)**

**Windows**: 10, **Node**: 12.11.1, **Compodoc**: 1.1.14

##### **Compodoc installed globally or locally ?**

Locally

##### **If possible sourcecode of the file where it breaks**

`import { DynamicFormsRoutes }                 from "app/core/components/patients-list/patients-list.routes.ts";`

##### **If possible your terminal logs before the error**

`[14:29:37] Analysing routes definitions and clean them if necessary`

##### **Motivation for or Use Case**

Because route generation is a feature of compodoc, but isn't usable if you're doing one of the most basic large project configuration changes in Angular, by including paths in your tsconfig, or using a different basepath.

##### **Related issues**

[ #403 Dec 7, 2017 - [BUG] baseUrl not taken in account ?](https://github.com/compodoc/compodoc/issues/403)
[ #885 Dec 13, 2010 - Project local paths defined in tsconfig.json are not getting picked up by compodoc[BUG]
](https://github.com/compodoc/compodoc/issues/885)

##### **Suggest a Fix**

Currently, there is one "reliable" fix for the problem, and that is by changing all of your routing imports to use relative paths 👎. This doesn't really work out well when you have hundreds of files being imported into a project.

A better long-term fix would be to apply the same logic that allows the path imports for all of the other documentation generations to be applied to the route doc generation.

**Comment by freed00m** [comment]
I also had a issue with path resolution in one of my configs with compodoc.

In the past I've created a workaround for one of the VScode plugin to correctly parse custom @myCustomPath/.. import resolutions

```
  /* The default angular '*.d.ts' glob pattern fails to parse
   *  the paths specified in the base.
   * Further development of the vscode-ng-language-service plugin
   *  might provide more robust parsing and code completion.
   **/
  "include": ["src/**/*.ts"],
```

Removing such include glob pattern makes the path resolution not fail for me.

**Comment by freed00m** [comment]
Duplicate of https://github.com/compodoc/compodoc/issues/1066

**Comment by Bengejd** [comment]
Fix submitted at #1148
